app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -346,15 +348,29 @@ private void setMapFragment() {
      * Calls fragment for list view.
      */
     private void setListFragment() {
-        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
-        Fragment fragment = new NearbyListFragment();
-        fragment.setArguments(bundle);
-        fragmentTransaction.replace(R.id.container, fragment, fragment.getClass().getSimpleName());
-        fragmentTransaction.commitAllowingStateLoss();
+        NearbyListFragment nearbyListFragment;
```

Comment:
Why are there changes in nearby activity?

Reply:
Hey @maskaravivek , first thing first:
1. Tests : Tests were failing since the fragment tag used in NearbyActivty and its corresponding tests had a mismatch, fixed that and upgraded robolectric to resolve a known issue with it (AndroidManifest not found). Now it passes all the tests, I have attached a screenshot of the same.

![tests](https://user-images.githubusercontent.com/20313518/36630273-b76ef868-1989-11e8-8d09-1ffc137e5706.png)

2. Changes in nearby activity: In nearby list fragment, setRetainInstance is set to true, so the fragment instance is retained during orientation change, but there is no condition to check whether fragment's instance already exists or not. So irrespective of existence of fragment's instance, nearby activity always creates a new instance during orientation change leading to slow loading and more network usage. Changes in nearyby activity fixes that.

3.Regarding issue, I opened another issue only after I pushed a fix for the current one. My bad, should have updated the issue thread also.

Reply:
Hi @knight-shade , you can absolutely create a new PR before the current one is merged. Simply create a new branch for your new fixes (I suggest creating it from master since your fixes aren't inter-related), and submit a new PR from there.

It is strongly advisable to submit separate PRs for separate fixes. ;)

Reply:
Thanks for the advice @misaochan . :+1:

Reply:
@knight-shade as @misaochan suggested create separate PRs for different fixes. The title or description has no mention of the other things that you have fixed. It would be great if you could create different PRs for the other fixes. :)

Also now there are some merge conflicts in this PR. You would have to resolve them too. :)

Reply:
On it @maskaravivek  :+1: